### DISCUSSION TITLE
[lit-html] How to disable dev mode?

### DISCUSSION THREAD

**Discussion by pruik** [lit/lit#3327] [Q&A]
We use `lit-html` to generate some HTML (custom components), which we will then run some unit tests on.
We continually receive warnings that we are using lit-html in dev mode, although this is not specified anywhere.

As an easy reproduction scenario, I created a little demonstration on codesandbox:
- https://codesandbox.io/s/lit-html-starter-forked-it4xsy?file=/src/index.ts

As you can see, the dev mode warning still shows up, even though I never defined development mode anywhere in the project.

How do I disable the dev mode?

EDIT: For what it's worth, version `1.4.1` does not throw the dev mode warning, anything above version `2.0.0` does.

**Comment by justinfagnani** [comment]
Dev mode has to be specifically _enabled_ - the default is the prod build. So there must be something in your tool chain that is using the `development` import condition and causing the dev build to be imported instead of the prod build.

On CodeSandbox I'm not sure what that would be. Maybe some bundler config, but don't see one in the files.

**Reply by pruik** [reply]
And that is exactly my point :upside_down_face: . I don't know what is causing the dev mode to trigger.

**Reply by augustjk** [reply]
What tool are using to build and/or serve? Look for options regarding export conditions. Sometimes tooling will have a "development" mode that sets a group of options together, including the export condition it uses.

**Reply by pruik** [reply]
Everything is on CodeSandbox, there's no export condition in there

**Reply by augustjk** [reply]
That sounds like a limitation of codesandbox that's out of our control then. It appears the sandbox you linked specifically uses parcel's dev server to show the preview pane which uses the development version.

If you must use codesandbox, perhaps you can use their [container environment](https://codesandbox.io/docs/environment#container-environment) to control the tooling.
